templates/hammer_root.yml.epp

```diff
@@ -1,8 +1,8 @@
 <%- |
   Optional[String] $username,
-  Optional[String] $password,
+  Optional[Variant[String, Sensitive[String]]] $password,
 | -%>
 :foreman:
   # Credentials. You'll be asked for the interactively if you leave them blank here
-  :username: '<%= $username %>'
-  :password: '<%= $password %>'
+  :username:<%= if $username { " '${username}'" } %>
+  :password:<%= if $password { " '${password}'" } %>
```

Comment:
Somehow this doesn't work for me. In #1018 I'm adding a test, but doesn't unwrap the string. I also checked if it perhaps was the interpolation, but that also doesn't appear to be it.